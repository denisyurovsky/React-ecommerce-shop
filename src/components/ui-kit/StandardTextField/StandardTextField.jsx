import { TextField } from '@mui/material';
import PropTypes from 'prop-types';
import React, { useState } from 'react';

export const StandardTextField = ({
  id,
  value,
  onChange,
  labelText,
  error,
  helperText,
  disabled,
  autoComplete,
}) => {
  const [label, setLabel] = useState(false);

  const onInputFieldFocus = () => {
    setLabel(true);
  };

  const onInputFieldBlur = () => {
    setLabel(false);
  };

  return (
    <TextField
      id={id}
      onChange={onChange}
      onFocus={onInputFieldFocus}
      onBlur={onInputFieldBlur}
      value={value}
      label={labelText}
      variant="standard"
      InputLabelProps={{
        shrink: Boolean(value.length) || label,
      }}
      margin="normal"
      error={error}
      disabled={disabled}
      autoComplete={autoComplete}
      helperText={helperText}
      fullWidth
    />
  );
};

StandardTextField.defaultProps = {
  id: null,
  autoComplete: 'on',
  error: false,
  helperText: '',
  disabled: false,
};

StandardTextField.propTypes = {
  id: PropTypes.string,
  error: PropTypes.bool,
  disabled: PropTypes.bool,
  autoComplete: PropTypes.oneOf(['on', 'off']),
  helperText: PropTypes.string,
  labelText: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
};
