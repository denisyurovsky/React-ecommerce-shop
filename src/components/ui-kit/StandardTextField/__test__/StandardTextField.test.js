import { render } from '@testing-library/react';
import React from 'react';

import { StandardTextField } from '../StandardTextField';

describe('StandardTextField component', () => {
  describe('snapshots', () => {
    it('renders a snapshot with all values', () => {
      const { asFragment } = render(
        <StandardTextField
          value="Value for tests"
          onChange={() => {}}
          labelText="Label for tests"
          error={false}
          helperText="Helper text fo tests"
          disabled={false}
          autoComplete="off"
        />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
