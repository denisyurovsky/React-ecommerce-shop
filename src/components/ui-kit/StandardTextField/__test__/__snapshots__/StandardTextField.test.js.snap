// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandardTextField component snapshots renders a snapshot with all values 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="mui-1"
      id="mui-1-label"
    >
      Label for tests
    </label>
    <div
      class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-ghsjzk-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-describedby="mui-1-helper-text"
        aria-invalid="false"
        autocomplete="off"
        class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id="mui-1"
        type="text"
        value="Value for tests"
      />
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-filled css-1d1r5q-MuiFormHelperText-root"
      id="mui-1-helper-text"
    >
      Helper text fo tests
    </p>
  </div>
</DocumentFragment>
`;
